test/functional/ui/messages_spec.lua

```diff
@@ -1628,32 +1629,24 @@ describe('ui/ext_messages', function()
       {1:~                        }Help poor children in Uganda!{1:                          }|
       {1:~                }type  :help iccf{5:<Enter>}       for information {1:                 }|
       {1:~                                                                               }|*5
-    ]])
+    ]]
+    screen:expect(introscreen)
 
     -- <c-l> (same as :mode) does _not_ clear intro message
     feed('<c-l>i')
-    screen:expect {
-      grid = [[
-      ^                                                                                |
-      {1:~                                                                               }|*4
-      {MATCH:.*}|
-      {1:~                                                                               }|
-      {1:~                 }Nvim is open source and freely distributable{1:                  }|
-      {1:~                           }https://neovim.io/#chat{1:                             }|
-      {1:~                                                                               }|
-      {1:~                }type  :help nvim{5:<Enter>}       if you are new! {1:                 }|
-      {1:~                }type  :checkhealth{5:<Enter>}     to optimize Nvim{1:                 }|
-      {1:~                }type  :q{5:<Enter>}               to exit         {1:                 }|
-      {1:~                }type  :help{5:<Enter>}            for help        {1:                 }|
-      {1:~                                                                               }|
-      {1:~{MATCH: +}}type  :help news{5:<Enter>} to see changes in v{MATCH:%d+%.%d+}{1:{MATCH: +}}|
-      {1:~                                                                               }|
-      {1:~                        }Help poor children in Uganda!{1:                          }|
-      {1:~                }type  :help iccf{5:<Enter>}       for information {1:                 }|
-      {1:~                                                                               }|*5
-    ]],
-      showmode = { { '-- INSERT --', 3 } },
-    }
+    screen:expect { grid = introscreen, showmode = { { '-- INSERT --', 3 } } }
+
+    -- opening a float also does not
+    local win = api.nvim_open_win(0, false, {
+      relative = 'editor',
+      height = 1,
+      width = 1,
+      row = 5,
+      col = 0,
+    })
+    command('mode')
+    screen:expect { unchanged = true, grid = introscreen, showmode = { { '-- INSERT --', 3 } } }
+    api.nvim_win_close(win, true)
```

Comment:
then, needs an assert that closing the window returns to the previous state that only the intro message was visible.

Otherwise, the `expect` below doesn't verify that specifically `feed('x')` was what hid the intro message anymore.